pkg/status/status.go

```diff
@@ -334,12 +282,33 @@ func (s *status) monitorClusterStatus() {
 	defer func() {
 		close(s.monitorDoneCh)
 	}()
+	startupFinished := false
 	for {
 		select {
 		case <-s.stopCh:
 			// If the stopCh is closed, the operator will exit and CO should
 			// be set to failing.
-			s.setFailing("Operator exited")
+			time := v1.Now()
+			statusConditions := []configv1.ClusterOperatorStatusCondition{{
```

Comment:
This seems like an overly verbose way of having to set the conditions. I think this can be modularized and will help easily indicate what you are setting here.

Reply:
I agree that creating 3 `ClusterOperatorStatusCondition` structures in an array is a bit verbose (@tkashem had brought this up as well). I had considered creating a builder function but that seemed like overkill. What are your thoughts on creating a function that returns  a `ClusterOperatorStatusCondition` object when provided `Type/Status/Message/LastTransactionTime` as arguments? This code would then be updated to create an array that call that function 3 times.

Reply:
Why is a builder an overkill? It might be a good fit here but it depends on how it ends up looking. You other suggestion seems like it would be doing what a builder does. Others might have other suggestions so please wait until you hear back from them too.

Reply:
@tkashem @kevinrizza thoughts?

Reply:
I think a builder is fine. A builder with methods like `SetFailing(bool)` or `SetAvailable(bool)` seems like a good pattern here. The useful change here is that you have separated setting those fields from actually passing them to the client.

Reply:
Yeah, a builder pattern works here. Another option is to use anonymous functions that can `closure` the local variables. The anonymous functions can be defined outside the `for {` loop.